megamek/src/megamek/common/Tank.java

```diff
@@ -580,59 +580,60 @@ public boolean isLocationProhibited(Coords c, int currElevation) {
 
         boolean hasFlotationHull = hasWorkingMisc(MiscType.F_FLOTATION_HULL);
         boolean isAmphibious = hasWorkingMisc(MiscType.F_FULLY_AMPHIBIOUS);
+        boolean hexHasRoad =  hex.containsTerrain(Terrains.ROAD);
 
         switch (movementMode) {
             case TRACKED:
                 if (!isSuperHeavy()) {
-                    return (hex.terrainLevel(Terrains.WOODS) > 1)
+                    return (hex.terrainLevel(Terrains.WOODS) > 1) && !hexHasRoad
```

Comment:
These and's and or's are confusing as to who binds to who, so I'm struggling to determine if these are "correct".

Reply:
I can put 'em into parens, and write up some comments.